wallet/src/main/java/com/android/identity_credential/wallet/PresentationActivity.kt

```diff
@@ -112,30 +111,42 @@ class PresentationActivity : FragmentActivity() {
             this.deviceEngagement = deviceEngagement
             Logger.i(TAG, "engagement info set")
 
-            launchPresentationActivity(context)
-            state.value = State.CONNECTED
+            if (phase.value != Phase.ENGAGING) {
+                launchPresentationActivity(context)
+            }
+            phase.value = Phase.CONNECTED
         }
 
         private fun launchPresentationActivity(context: Context) {
             val launchAppIntent = Intent(context, PresentationActivity::class.java)
             launchAppIntent.addFlags(
                 Intent.FLAG_ACTIVITY_NEW_TASK or
                         Intent.FLAG_ACTIVITY_NO_HISTORY or
-                        Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS
+                        Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS or
+                        Intent.FLAG_ACTIVITY_NO_ANIMATION
             )
             context.startActivity(launchAppIntent)
         }
 
         fun isPresentationActive(): Boolean {
-            return state.value != State.NOT_CONNECTED
+            return phase.value != Phase.NOT_CONNECTED
+        }
+
+        fun stopPresentationReaderTimeout(context: Context) {
+            resultStringId = R.string.presentation_result_error_message_reader_timeout
+            resultDrawableId = R.drawable.presentment_result_status_error
+            phase.value = Phase.SHOW_RESULT
         }
     }
 
-    enum class State {
+    enum class Phase {
```

Comment:
👏 I feel like this class had it coming.. it was a matter of time before it was changed to a more meaningful state of the PresentationActivity